tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -773,6 +793,21 @@ private String coalesceLiterals(StringBuilder l1, StringBuilder l2) {
             return result;
         }
 
+        private boolean endsWith(StringBuilder l1, StringBuilder l2) {
+            final int l2Length = l2.length();
+            final int delta = l1.length() - l2Length;
+            if (delta < 0) {
+                return false;
+            }
+            for (int i = 0; i < l2Length; ++i) {
+                // don't have to worry about unpaired surrogates.
```

Comment:
why, out of curiosity?

Reply:
DAIP does "normalize all input to NFC" so how could we get an unpaired surrogate?

Reply:
denormalized would be a different issue, that would be something like `"é".endsWith("\u0301")`

Reply:
I was imagining that com.ibm.icu.text.Normalizer#compose() would never return a string containing an unpaired surrogate; I could be wrong...

Reply:
I was wrong! Just tested like this with an unpaired surrogate:
```
        String xValue = Normalizer.compose("\uDC04", false);
        System.out.println("xValue = " + xValue + " " + xValue.contains("\uDC04"));
```
[INFO] xValue = ? true

It may not be likely in practice, still it's worrisome ... we should add a check to DAIP for well-formed strings!

Reply:
it could be tested like this:
```
        CharsetEncoder encoder = Charset.forName("UTF-16BE").newEncoder();
        try {
            char[] data = value.toCharArray();
            encoder.encode(CharBuffer.wrap(data));
        } catch (Exception e) {
            internalException[0] = e;
            return null;
        }
```

Reply:
Right, as long as values are well-formed, it doesn't matter. We could file a ticket to raise an error for vetters for any value that was ill-formed. I wouldn't make it a high priority, and if we do this we can do it more efficiently than firing up an encoder, because the rules are so simple (ICU might have a method already).

Reply:
OK so it's really:

```suggestion
                // not currently worrying about unpaired surrogates.
```

anyway, not a problem